### PR TITLE
Switched the registries to C++20's designated initializers.

### PR DESCRIPTION
**Pull request by VZout**
* Registries now get destroyed properly.
* Removed the named type from the registries in favor of designated initialization.
* Now using std's `contains` function instead of my own.

### CODE REVIEW

File: src/engine_registry.cpp [hunk 1]
```diff
@@ -31,11 +31,11 @@ namespace wr
 		);
 
 	REGISTER(root_signatures::brdf_lut, RootSignatureRegistry)({
-		RootSignatureDescription::Parameters({
+		.m_parameters = {
```
Comment (DBouma) [inline src/engine_registry.cpp]: Not entirely sure what the . prefix does here, same goes for the other prefixes, it is explicit and clear though but might be confusing for the team

File: src/engine_registry.cpp [hunk 2]
```diff
@@ -75,331 +75,329 @@ namespace wr
...
-		PipelineDescription::TopologyType(TopologyType::TRIANGLE)
-		}
-	);
+		.m_vertex_shader_handle = std::nullopt,
+		.m_pixel_shader_handle = std::nullopt,
+		.m_compute_shader_handle = shaders::brdf_lut_cs,
+		.m_root_signature_handle = root_signatures::brdf_lut,
+		.m_dsv_format = Format::UNKNOWN,
+		.m_rtv_formats = { Format::R16G16_FLOAT },
+		.m_num_rtv_formats = 1,
```
Comment (DBouma) [inline src/engine_registry.cpp]: m_num_rtv_formats is a uint, might be a good idea to explicitly assign an unsigned value. Not too important though, just consistancy as we do this for most of the project.

File: src/engine_registry.cpp [hunk 3]
```diff
@@ -444,441 +442,441 @@ namespace wr
...
 
 	// Cone of confusion
 	REGISTER(shaders::dof_coc, ShaderRegistry) ({
-		ShaderDescription::Path("resources/shaders/dof_coc.hlsl"),
-			ShaderDescription::Entry("main_cs"),
-			ShaderDescription::Type(ShaderType::DIRECT_COMPUTE_SHADER)
-		});
+		.path = "resources/shaders/dof_coc.hlsl",
+		.entry = "main_cs",
+		.type = ShaderType::DIRECT_COMPUTE_SHADER
```
Comment (DBouma) [inline src/engine_registry.cpp]: Perhaps make this variable more explicit? Like ".shader_type", just a suggestion

File: src/pipeline_registry.hpp [hunk 4]
```diff
@@ -16,37 +16,22 @@ namespace wr
...
-
-		InputLayout m_input_layout = InputLayout({});
+		std::optional<RegistryHandle> m_vertex_shader_handle;
+		std::optional<RegistryHandle> m_pixel_shader_handle;
+		std::optional<RegistryHandle> m_compute_shader_handle;
+		RegistryHandle m_root_signature_handle;
+
+		Format m_dsv_format;
+		std::array<Format, 8> m_rtv_formats;
+		unsigned int m_num_rtv_formats;
```
Comment (DBouma) [inline src/pipeline_registry.hpp]: Perhaps declare this as std::uint32_t for consistancy?

**Comment by VZout** [comment]
Merging, its not perfect but nothing related to the changes itself. the system wasn't perfect to begin with.

**Pull request merged**